File: android/src/androidTest/java/org/ligi/passandroid/BaseUnitTest.java
package org.ligi.passandroid;

import android.support.test.InstrumentationRegistry;
import org.ligi.passandroid.reporting.SpooningFailureHandler;
import static android.support.test.espresso.Espresso.setFailureHandler;


public abstract class BaseUnitTest {

    public void setUp()  {
        setFailureHandler(new SpooningFailureHandler(InstrumentationRegistry.getInstrumentation()));
    }

}

File: android/src/androidTest/java/org/ligi/passandroid/BaseUnitTest.kt
package org.ligi.passandroid

import android.app.Activity
import android.support.test.InstrumentationRegistry
import android.support.test.espresso.Espresso.setFailureHandler
import android.view.WindowManager
import org.ligi.passandroid.reporting.SpooningFailureHandler


abstract class BaseUnitTest {

    fun setUp(activity: Activity) {
        setFailureHandler(SpooningFailureHandler(InstrumentationRegistry.getInstrumentation()))

        activity.runOnUiThread { activity.window.addFlags(WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON or WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD) }
    }

}
